etcd.go

```diff
@@ -0,0 +1,127 @@
+package dsync
+
+import (
+	"github.com/coreos/etcd/client"
+	"github.com/pborman/uuid"
+	"golang.org/x/net/context"
+	"sync"
+	"time"
+)
+
+type etcdMutex struct {
+	backoff BackoffFunc
+	ctx     context.Context
+	etcd    client.KeysAPI
+	key     string
+	locked  bool
+	m       sync.Mutex
+	refresh time.Duration
+	stop    chan struct{}
+	ttl     time.Duration
+	uuid    string
+	wg      sync.WaitGroup
+}
+
+func (e *etcdMutex) Lock() {
+	var i int64
+	e.m.Lock()
+	for e.locked != false {
+		// The lock is already held within this process.
+		// Wait for the lock to be unlocked.
+	}
+	e.stop = make(chan struct{})
+	// Error if the previous value of the key is not empty.
+	var options = client.SetOptions{PrevExist: client.PrevNoExist, TTL: e.ttl}
```

Comment:
Use var to declare a variable that will be set to its zero value. In this case you want to use the :=